File: FUNDAMENTAL JavaScript/Regular Expressions/furniture.js
function furniture(input) { 

    let totalSum = 0;

    console.log("Bought furniture:")
    for (let line of input) {
        const pattern = />>(?<name>.+)<<(?<price>[\d]+\.?[\d]+)!(?<quantity>[\d]+)/g;
        
        const matches = pattern.exec(line);

        if(matches) {
            let {name, price, quantity} = matches.groups;
            
            console.log(name);
            totalSum += (Number(price)) * (Number(quantity));

        }
        
    }
    console.log(`Total money spend: ${totalSum.toFixed(2)}`);
}

furniture([">>Sofa<<312.23!3",
    ">>TV<<300!5",
    ">Invalid<<!5",
    "Purchase"]);

File: Java Fundamentel/src/regularEpressions/exercise/Furniture.java
package regularEpressions.exercise;

import java.util.Scanner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class Furniture {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String input = scanner.nextLine();

        double totalSum = 0.0;

        StringBuilder sb = new StringBuilder("Bought furniture:").append(System.lineSeparator());
        String regex = ">>(?<item>[A-Za-z]+)<<(?<price>[0-9]+\\.?[0-9]*)!(?<quantity>[0-9]+)";
        Pattern pattern = Pattern.compile(regex);

        while (!input.equals("Purchase")) {
            Matcher matcher = pattern.matcher(input);
            if (matcher.find()) {
                sb.append(matcher.group("item")).append(System.lineSeparator());
                totalSum += Double.parseDouble(matcher.group("price")) * Double.parseDouble(matcher.group("quantity"));
            }
            input = scanner.nextLine();
        }
        sb.append("Total money spend: ");
        sb.append(String.format("%.2f", totalSum));

        System.out.println(sb.toString());

    }
}
